File: app/src/main/java/com/kirchhoff/example/githubclient/api/GitHubService.java
package com.kirchhoff.example.githubclient.api;

import com.google.gson.JsonObject;
import com.kirchhoff.example.githubclient.model.Authorization;
import com.kirchhoff.example.githubclient.model.CommitResponse;
import com.kirchhoff.example.githubclient.model.Repository;

import java.util.List;

import retrofit2.http.Body;
import retrofit2.http.GET;
import retrofit2.http.Header;
import retrofit2.http.POST;
import retrofit2.http.Path;
import rx.Observable;

/**
 * @author Kirchhoff-
 */

public interface GitHubService {

    @POST("/authorization")
    Observable<Authorization> authorize(@Header("Authorization") String authorization,
                                        @Body JsonObject params);

    @GET("/user/repos")
    Observable<List<Repository>> repositories();

    @GET("/repos/{user}/{repo}/commits")
    Observable<List<CommitResponse>> commits(@Path("user") String user, @Path("repo") String repo);
}

File: app/src/main/java/com/kirchhoff/example/githubclient/api/GitHubService.kt
package com.kirchhoff.example.githubclient.api

import com.google.gson.JsonObject
import com.kirchhoff.example.githubclient.model.Authorization
import com.kirchhoff.example.githubclient.model.CommitResponse
import com.kirchhoff.example.githubclient.model.Repository
import retrofit2.http.*
import rx.Observable

/**
 * @author Kirchhoff-
 */

interface GitHubService {

    @POST("/authorizations")
    fun authorize(@Header("Authorization") authorization: String,
                  @Body params: JsonObject): Observable<Authorization>

    @GET("/user/repos")
    fun repositories(): Observable<List<Repository>>

    @GET("/repos/{user}/{repo}/commits")
    fun commits(@Path("user") user: String, @Path("repo") repo: String)
            : Observable<List<CommitResponse>>
}
